### PR TITLE
Release: 4.1.0

### PR DESCRIPTION
**Pull request by aboedo**
#### API updates: 

- Added new method `checkTrialOrIntroDiscountEligibility(product:)`, which allows you to check for intro or discount eligibility for a single `StoreProduct`. 
    https://github.com/RevenueCat/purchases-ios/pull/1354

- Added explicit parameter name for `checkTrialOrIntroDiscountEligibility(productIdentifiers:)`. 
The existing method without the parameter name still work, but is now deprecated. Xcode will offer an auto fix-it for it. 
    https://github.com/RevenueCat/purchases-ios/pull/1362

- Made `StoreProduct` initializers public so they can be used for testing. 

#### Other:

- Added auto-fix-it for `invalidatePurchaserInfoCache` rename
    https://github.com/RevenueCat/purchases-ios/pull/1379

- Docs improvements
- A lot of under-the-hood improvements, mainly focusing on network requests and tests.

## Requirements: 
- https://github.com/RevenueCat/purchases-ios/pull/1381

### CODE REVIEW

File: IntegrationTests/CommonFiles/RevenueCat-Swift.h [hunk 1]
```diff
@@ -2143,6 +2183,12 @@ SWIFT_CLASS_NAMED("StoreTransaction")
 
 
 
+@interface RCStoreTransaction (SWIFT_EXTENSION(RevenueCat))
+@property (nonatomic, readonly, copy) NSString * _Nonnull productId SWIFT_AVAILABILITY(macos,obsoleted=1,message="'productId' has been renamed to 'productIdentifier'") SWIFT_AVAILABILITY(watchos,obsoleted=1,message="'productId' has been renamed to 'productIdentifier'") SWIFT_AVAILABILITY(tvos,obsoleted=1,message="'productId' has been renamed to 'productIdentifier'") SWIFT_AVAILABILITY(ios,obsoleted=1,message="'productId' has been renamed to 'productIdentifier'");
+@property (nonatomic, readonly, copy) NSString * _Nonnull revenueCatId SWIFT_AVAILABILITY(macos,obsoleted=1,message="'revenueCatId' has been renamed to 'transactionIdentifier'") SWIFT_AVAILABILITY(watchos,obsoleted=1,message="'revenueCatId' has been renamed to 'transactionIdentifier'") SWIFT_AVAILABILITY(tvos,obsoleted=1,message="'revenueCatId' has been renamed to 'transactionIdentifier'") SWIFT_AVAILABILITY(ios,obsoleted=1,message="'revenueCatId' has been renamed to 'transactionIdentifier'");
+@end
+
+
```
Comment (aboedo) [inline IntegrationTests/CommonFiles/RevenueCat-Swift.h:2191]: docs updates begin here, you can skip everything from here until my next comment

File: Purchases/Info.plist [hunk 2]
```diff
@@ -17,7 +17,7 @@
 	<key>CFBundlePackageType</key>
 	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
-	<string>4.0.0</string>
```
Comment (aboedo) [inline Purchases/Info.plist:20]: docs updates end here, from here on you can review again

**Comment by aboedo** [comment]
This _looks_ like a lot of file changes but a lot of it is just the docs update. This'll get fixed in #1368

File: Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -61,7 +61,7 @@
 		5576344125A3C06E00C0486A /* Views */ = {
 			isa = PBXGroup;
 			children = (
-				5576344225A3C08400C0486A /* PackageCell.swift */,
+				5576344225A3C084.1.0486A /* PackageCell.swift */,
```
Comment (NachoSoto) [inline Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj:64]: This seems broken? 🤔

**Review by NachoSoto — APPROVED** [review]
😻

File: Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj (hunk 3 above)
Comment (aboedo) [reply Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj:64]: lol I suppose `sed` must have caught that 400?? I'm not even sure how that happened, I'll revert

Comment (aboedo) [reply Examples/MagicWeather/MagicWeather.xcodeproj/project.pbxproj]: fixed

**Pull request merged**